lib/Frontend/CompilerInvocation.cpp

```diff
@@ -1822,10 +1822,14 @@ static bool ParseIRGenArgs(IRGenOptions &Opts, ArgList &Args,
                                           RenderedArgs, SDKPath,
                                           ResourceDir);
     }
-    // TODO: Should we support -fdebug-compilation-dir?
-    llvm::SmallString<256> cwd;
-    llvm::sys::fs::current_path(cwd);
-    Opts.DebugCompilationDir = std::string(cwd.str());
+
+    if (const Arg *A = Args.getLastArg(OPT_debug_compilation_dir))
+      Opts.DebugCompilationDir = std::string(A->getValue());
```

Comment:
is the explicit `std::string()` constructor actually necessary?

Reply:
Frankly speaking, I am not sure.  It still works if I remove the constructor. Do you recommend to remove it if either way works?

Reply:
Yes, please!